File: ALU_Garage_Maintainer/ALU_WebApp/src/app/Interfaces/Car.ts
export interface ICar{
    class: string;
    rarity: string;
    maxStars: number;
    maxRank: number;
    name: string;
    fuel: number;
    topSpeed: number;
    acceleration: number;
    handling: number;
    nitro: number;
    hasEips: boolean;
    noEips: number | null;
    requiresKey: boolean;
    bp1sCount: number|null;
    bp2sCount: number|null;
    bp3sCount: number|null;
    bp4sCount: number|null;
    bp5sCount: number|null;
    bp6sCount: number|null;
}

File: ALU_Garage_Maintainer/ALU_DAL/Models/Car.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace ALU_DAL.Models;

public partial class Car
{
    public int Id { get; set; }

    public string Name { get; set; }

    public string Class { get; set; }

    public byte Fuel { get; set; }

    public string Rarity { get; set; }

    public decimal TopSpeed { get; set; }

    public decimal Acceleration { get; set; }

    public decimal Handling { get; set; }

    public decimal Nitro { get; set; }

    public byte MaxStars { get; set; }

    public decimal MaxRank { get; set; }

    public bool HasEips { get; set; }

    public byte? NoEips { get; set; }

    public bool RequiresKey { get; set; }

    public int? Bp1sCount { get; set; }

    public int Bp2sCount { get; set; }

    public int Bp3sCount { get; set; }

    public int? Bp4sCount { get; set; }

    public int? Bp5sCount { get; set; }

    public int? Bp6sCount { get; set; }

    [JsonIgnore]
    public virtual Class ClassNavigation { get; set; }
}
